### DISCUSSION TITLE
module dialogs are now embedded in the options dialog

### DISCUSSION THREAD

**Discussion by sthalik** [opentrack/opentrack#1393] [Ideas]
As you know, the opentrack main window is awfully cluttered. I took the liberty to change it and made the options dialog cluttered instead. Things now look like this:

![image](https://user-images.githubusercontent.com/1896811/146589088-73e3b278-93e9-4bb9-8be2-8ecb5ae686ef.png)
![image](https://user-images.githubusercontent.com/1896811/146589114-176324da-8def-4a9c-af11-f31703522274.png)

Dialogs that haven't yet been converted display in their usual form.

This has the effect of breaking the workflow for existing opentrack users. How much do you hate this? A lot, or only a little? Should I provide an opt-out switch in the options dialog?

sh

**Comment by sthalik** [comment]
@GO63-samara @Slion @DaWelter how much do you hate this change? Should it be adopted by all modules?

**Comment by GO63-samara** [comment]
I have checked these changes in `OT-20191220_01`.
I love the way it looks and how it works.
I would add the name of the selected tracker and the selected filter to the `Options-Tracker` and `Options-Filter` panel, so it will be clearer.
These changes have not yet been added for other trackers and filters. This is quite a lot of work to do.
And some old users may also grumble about these changes.

**Comment by sthalik** [comment]
I'm not sure whether it's better. It should be easier to navigate across profiles if the names are always uniform. The names are of varying width so clicking at the right place doesn't work well with muscle memory and thus violates the POLA.

![image](https://user-images.githubusercontent.com/1896811/147889648-b1e8fb28-a9ed-4de1-a443-0c89190ea435.png)
[module-names.patch.txt](https://github.com/opentrack/opentrack/files/7799811/module-names.patch.txt)

**Comment by GO63-samara** [comment]
Then maybe it would be better to display the name of the filter in the settings panel?

![image](https://user-images.githubusercontent.com/63638386/147917088-1b86b82b-5e7d-4d77-a2ac-1bf9a51ab019.png)

I wonder what will be displayed if the filter is disabled? This mode is also needed for work.

**Reply by sthalik** [reply]
The modules are only selected on the main window. This is preferable because adding a dynamic tab to the options dialog has to reflow the entire dialog. I don't think I properly understood your question?

One case that isn't handled right now is adding a module in another's stead (i.e. changing from Accela to Hamilton) while the options dialog is already open. What's going to happen is settings for that tab becoming reverted and the tab simply disappearing. It's only going to be created once the entire options dialog is opened again.

The solutions to these issues are pretty clear, but I'd rather hear user feedback before working on an overengineered solution.

**Comment by GO63-samara** [comment]
I meant not to indicate the name of the tracker on the label, but to indicate the name of the tracker under the label, in the settings panel. Then the width of the label will not change when the tracker is changed.

All of these changes will only remove the hammer buttons from the main window, right?

**Comment by sthalik** [comment]
The hammer buttons should stay as windows, tabs and menus don't work well with muscle memory. It only makes sense to remove them in case where module selection on the main window is removed entirely.

It makes sense to add a title from an aesthetics standpoint, but the window is already having trouble accomodating all the vertical space. Also, it makes tab switch handling and pointer ownership harder. I'd rather avoid complicating the already tortured logic unless it's very frequently requested.
